lib/astutils.cpp

```diff
@@ -2571,7 +2571,7 @@ bool isVariableChanged(const Token *tok, int indirect, const Settings *settings,
     tok2 = skipRedundantPtrOp(tok2, tok2->astParent());
 
     if (tok2->astParent() && tok2->astParent()->isAssignmentOp()) {
-        if ((indirect == 0 || tok2 != tok) && tok2 == tok2->astParent()->astOperand1())
+        if (((indirect == 0 || tok2 != tok) || (indirect == 1 && tok2->str() == ".")) && tok2 == tok2->astParent()->astOperand1())
```

Comment:
Shouldn't this be `tok2->originalName() == "->"`?

Reply:
Could be. Do you have a test case?
It's weird that these exceptions are needed in the first place.

Reply:
Well in the test cases it uses `s->x`(which unfortunately gets simplified to `s.x`) so I assume thats why you added it for `indirect == 1`, which usually means dereferencing the pointer changes the variable(which is what we are doing in this case). This wouldn't be the case for `s.x` in the code, so it would make sense to only check for `->`.

Reply:
I couldn't find an example where this makes a difference. We already check that there is an assignment, so something (member, parent, both?) is changed.